source/game/system/GhostFile.cc

```diff
@@ -181,6 +195,13 @@ bool RawGhostFile::isValid(const u8 *rkg) const {
     return true;
 }
 
+u32 RawGhostFile::readRKGTime(const u8 *timeBytes) const {
+    u32 minutes = *timeBytes; // The minutes are doubled here
```

Comment:
mmmmm I don't think the minutes are doubled here. What I think is actually happening is this assumes minutes is 8 bits. In reality it's 7 bits, which means you are missing a `>> 1` here, hence why the minutes appears doubled. This dangling bit is actually the MSB of `seconds`.

Anyway, we already have this implemented. Earlier in `GhostFile.cc` we parse out lap times (see line 47). Since we don't use a stream in `RawGhostFile::isValid` you can instead do:
```c++
Timer lapTime = Timer(parse<u32>(*reinterpret_cast<const u32*>(timeBytes)));
```
And then we overloaded `operator+` so you can just have another timer for the sum and do `lapTimeSum += lapTime`

Reply:
as explained on discord